pkg/client/cache/listwatch.go

```diff
@@ -31,16 +32,16 @@ import (
 type ListerWatcher interface {
 	// List should return a list type object; the Items field will be extracted, and the
 	// ResourceVersion field will be used to start the watch in the right place.
-	List(options api.ListOptions) (runtime.Object, error)
+	List(options v1.ListOptions) (runtime.Object, error)
```

Comment:
> except that right now we're introducing a change from internal to versioned that ripples out into other code

IIUC, "ripples" refers to all the calls to `selector.String()`? Keeping ListWatcher interface taking api.ListOptions will only reduce very limited amount of the ripples, because most ripples are not caused by it, but rather because the versioned clientset's [List/Watch methods](https://github.com/kubernetes/kubernetes/blob/master/pkg/client/clientset_generated/release_1_5/typed/core/v1/pod.go#L40-L41) take v1.ListOptions. Only a few ListFunc/WatchFunc definitions refer to the selectors, like this [one](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/client_builder.go#L114).

So if we really want to reduce the ripples, we need to revert https://github.com/kubernetes/kubernetes/pull/31994. Otherwise there's not point keeping this interface using api.ListOptions, it will just introduce calls to conversion functions in all ListFunc/WatchFunc definitions.